include/AnalogSensor.h

```diff
@@ -7,10 +7,13 @@
 class AnalogSensor : public Sensor
 {
     uint8_t pin = 0;
+    unsigned long long runner = 0; // Lots of buffer
+    unsigned long counter = 0;
 public:
     AnalogSensor(ReservedIDs id, bool criticality, uint8_t pin, uint32_t readInterval);
-    [[nodiscard]] Health healthCheck() const override;
```

Comment:
Correct. Has to read and save the data from the serial line to determine if a legible NMEA sentence has been sent

Reply:
Would it be possible to put this into a function *not* belonging to the class?

For example:

```cpp
void healthCheck(*Adafruit_GPS gps) {
    /// Health check stuff
}

class GPS final : public Sensor {
    Health healthCheck() {
        healthCheck(this->wGPS)
    }
}

```

This would make it so we don't have to change as many files and keep const correctness

Reply:
It might not make a lot of sense in the context if you need a lot of variables, but might be worth considering

Reply:
possible, maybe, but tbh to me that is really confusing then just not having be a constant because it needs to not be const and it is associated with the class.

Reply:
I agree. Would be nice if it's const but not worth